cmake/options.cmake

```diff
@@ -1,6 +1,9 @@
 # Copyright (c) Microsoft Corporation
 # SPDX-License-Identifier: MIT
 
+set(CMAKE_SYSTEM_VERSION 10.0.19041.0)
+set(CMAKE_CXX_STANDARD 20)
```

Comment:
For consistency, I think this can be moved under `cmake/settings.cmake`. Thanks to `CMAKE_CXX_STANDARD` we can also remove the following lines at the end:

```cmake
target_compile_features("ebpf_for_windows_cpp_settings" INTERFACE
  cxx_std_17
)
```

Reply:
It appears as if the CMAKE_SYSTEM_VERSION directive has to be before the first project directive for it to work. Moved the C++ version directive to the settings.cmake file.